### DISCUSSION TITLE
Home row mods and one shot keys

### DISCUSSION THREAD

**Discussion by dafzor** [sezanzeb/input-remapper#991] [Q&A]
Been trying out [home row mods](https://precondition.github.io/home-row-mods) with input remapper but can't get the detection to actually work reliably.

Some investigation showed that the solution would be to use press time to detect tap vs hold, but can't seem to see any way to do that with the current macro system.

Another thing i'm also exploring is one shot mods where you press a key and only the next key is modified (example: press shift let go, then a and b and only A gets capitalized). I can do that with the macro system but would have to modify every key in the keyboard to use a set and if_eq.

So question is if I'm missing something that would make what i'm trying to do easier?

**Comment by sezanzeb** [comment]
The `mod_tap` macro is available on the main branch now for anyone to install. I'll probably make a new proper release that includes this in a few weeks.
